components/renku_data_services/crc/orm.py

```diff
@@ -152,15 +156,21 @@ def dump(self, quota: models.Quota | None) -> models.ResourcePool:
                 detail=f"The quota name in the database {self.quota} and Kubernetes {quota.id} do not match.",
             )
         if (quota is None and self.quota is not None) or (quota is not None and self.quota is None):
-            raise errors.BaseError(
-                message="Unexpected error when dumping a resource pool ORM.",
-                detail=f"The quota in the database {self.quota} and Kubernetes {quota} do not match.",
+            logging.error(
+                f"Unexpected error when dumping resource pool ORM with ID {self.id}. "
+                f"The quota in the database {self.quota} and Kubernetes {quota} do not match. "
+                f"Using the quota {quota} in the response."
             )
         return models.ResourcePool(
             id=self.id,
             name=self.name,
             quota=quota,
-            classes=[resource_class.dump() for resource_class in classes],
+            classes=[
+                resource_class.dump(
+                    resource_class.dump() >= class_match_criteria if class_match_criteria is not None else None
+                )
```

Comment:
**question (non-blocking):** this looks weird to me, we dump the class twice, once for matching and then once for the output. could this be done in one go by passing the class_match_criteria into the dump method or by dumping first and then setting `matching` afterwards?

Reply:
Yeah I mean to go back to this and clean it up.